src/core/domain-services/remote-key-service.ts

```diff
@@ -71,7 +71,8 @@ export class RemoteKeyHost {
             console.log("Inside RemoteKeyHost::in::Msg, senderId: ", msg, senderId);
             const data = await this.handleMessage(msg);
             console.log("Inside RemoteKeyHost::initialize::Data(handleMessage): ", data);
-            this.sendInvocationResult(data, senderId!);
+            // @ts-ignore
+            await this.sendInvocationResult(data, CruxId.fromString(senderId!)); // getting senderId as string
```

Comment:
ID should be coming as object here. If it isn't then it's a bug! Can you fix it?
SecureReceiveSocket will be the right place to do it. Please think about why it's the right place to do it, let me know if you disagree.

Reply:
yeah seems so. will check it again and fix it. And yes SecureRecieveSocket will be best as it will make sure that it's consistent throughout.